src/System.Buffers/src/System.Buffers.csproj

```diff
@@ -11,13 +11,14 @@
     <AssemblyVersion Condition="$([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) == '.NETFramework'">4.0.3.0</AssemblyVersion>
     <AssemblyVersion Condition="'$(IsPackable)' == 'true' and $([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) == '.NETFramework'">4.0.4.0</AssemblyVersion>
     <PackageValidationBaselineVersion>4.5.1</PackageValidationBaselineVersion>
+    <IsPlaceholderTargetFramework Condition="'$(TargetFramework)' == 'netcoreapp2.1'">true</IsPlaceholderTargetFramework>
   </PropertyGroup>
 
-  <PropertyGroup Condition="$([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) == '.NETCoreApp'">
+  <PropertyGroup Condition="'$(IsPlaceholderTargetFramework)' == 'true'">
     <EnableDefaultItems>false</EnableDefaultItems>
```

Comment:
This setting should be the default for placeholder TFMs. Move it into the targets file.

Reply:
Can I have a property group in Directory.Build.targets before the Imports section?
Shouldn't I add it to Directory.Build.props instead?

Reply:
I had to add it to Directory.Build.props. If I add it to the targets file, the property is not evaluated on time.

Reply:
I didn't mean the `IsPlaceholderTargetFramework` property. That one needs to stay in the project files as it shouldn't apply to all projects, just to some.

I meant that the `EnableDefaultItems` property should be moved into the new `placeholderpackaging.targets` file.

Reply:
Ah I see. Note that Microsoft.Bcl.HashCode enables the setting too, but with a different condition, so that one will stay in place:

```xml
<PropertyGroup Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netstandard2.1'))">
  <EnableDefaultItems>false</EnableDefaultItems>
</PropertyGroup>
```

Reply:
If `IsPlaceholderTargetFramework` is defined directly in the csproj, but `EnableDefaultItems` is set to false in placeholderpackaging.targets (conditioned by `IsPlaceholderTargetFramework == true`), then `EnableDefaultItems` never gets evaluated because `IsPlaceholderTargetFramework` is not yet defined at that point.

I also tried putting it in Directory.Build.props or in Directory.Build.targets, but they also don't set the property on time.

What am I doing wrong?

Reply:
> If IsPlaceholderTargetFramework is defined directly in the csproj, but EnableDefaultItems is set to false in placeholderpackaging.targets (conditioned by IsPlaceholderTargetFramework == true), then EnableDefaultItems never gets evaluated because IsPlaceholderTargetFramework is not yet defined at that point.

But that sounds right. Placeholderpackaging.targets is imported after the project file in the root Directory.Build.targets file so it shoul already have access to the `IsPlaceholderTargetFramework` property.

Reply:
Fixed. Should remove IsPackable conditioning when importing the placeholderpackaging.targets file.